### PR TITLE
Deterministic Deploys

### PR DESCRIPTION
**Pull request by AStox**

### CODE REVIEW

**Comment by offerijns** [comment]
@AStox I noticed the env vars were not set in CI so added those. And while I was at it, also fixed two tests (that were broken in a previous PR) and formatting.

File: test/token/factory.t.sol [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        uint8 decimals
+    ) public {
+        vm.selectFork(mainnetFork);
+        TrancheTokenFactory tokenFactory1 = new TrancheTokenFactory{ salt: SALT }();
+        address token1 = tokenFactory1.newTrancheToken(poolId, trancheId, name, symbol, decimals);
+
+        vm.selectFork(polygonFork);
+        TrancheTokenFactory tokenFactory2 = new TrancheTokenFactory{ salt: SALT }();
+        assertEq(address(tokenFactory1), address(tokenFactory2));
+        vm.prank(sender);
```
Comment (offerijns) [inline test/token/factory.t.sol:35]: Is there a reason this `vm.prank` is not higher up in the test, so it's also a different sender when deploying the factories?

File: test/token/factory.t.sol [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        uint256 threshold
+    ) public {
+        vm.selectFork(mainnetFork);
+        MemberlistFactory memberlistFactory1 = new MemberlistFactory{ salt: SALT }();
+        address memberlist1 = memberlistFactory1.newMemberlist();
+
+        vm.selectFork(polygonFork);
+        MemberlistFactory memberlistFactory2 = new MemberlistFactory{ salt: SALT }();
+        assertEq(address(memberlistFactory1), address(memberlistFactory2));
+        vm.prank(sender);
```
Comment (offerijns) [inline test/token/factory.t.sol:130]: Likewise

**Review by offerijns — DISMISSED** [review]
Great idea to use the forks, and nice tests 👍

File: test/token/factory.t.sol (hunk 1 above)
Comment (NunoAlexandre) [reply test/token/factory.t.sol:35]: Same question and TIL on `vm.prank` :D

File: test/Connector.t.sol [hunk 3]
```diff
@@ -373,7 +373,7 @@ contract ConnectorTest is Test {
         address destinationAddress,
         uint128 amount
     ) public {
-        vm.assume(validUntil > block.timestamp + 7 days);
+        vm.assume(validUntil > block.timestamp + 7 days && destinationAddress != address(0));
```
Comment (NunoAlexandre) [inline test/Connector.t.sol:376]:
```suggestion
        vm.assume(validUntil > block.timestamp + 7 days);
        vm.assume(destinationAddress != address(0)));
```

**Review by NunoAlexandre — COMMENTED** [review]
Just one small thing. Really nice to have these tests in place ✨

File: test/Connector.t.sol (hunk 3 above)
Comment (offerijns) [reply test/Connector.t.sol:376]: Good point that's cleaner.

File: test/token/factory.t.sol (hunk 1 above)
Comment (AStox) [reply test/token/factory.t.sol:35]: Well the factories addresses will be affected by the sender, but we expect that the created token contracts shouldn't be affected by the sender address, so by putting the prank here we're also testing that anybody can deploy a token from a factory and expect the address to be the same.

Comment (offerijns) [reply test/token/factory.t.sol:35]: Right makes a lot of sense 👍

**Review by offerijns — APPROVED** [review]

**Review by offerijns — APPROVED** [review]

**Pull request merged**